Minor architecture docs tweaks

awly opened the pull request:
- consistently use "certificate" instead of "public key"
- make diagram in "local users" section match the text (user "sandra"
  doesn't have access to "grav-02")
- de-duplicate docs on session streaming between auth and proxy pages

benarent commented on docs/4.2/architecture/teleport_architecture_overview.md:
```diff
@@ -219,11 +218,11 @@ $ tsh --proxy=p scp -P example.txt user@host/destination/dir
-When `tsh` logs in, the auto-expiring key is stored in `~/.tsh` and is valid for
-12 hours by default, unless you specify another interval via the `--ttl` flag
-(capped by the server-side configuration).
+When `tsh` logs in, the auto-expiring certificate is stored in `~/.tsh` and is
```
It's good to have some fresh eyes on these docs. +1 for using certificates vs keys

benarent commented on docs/4.2/admin-guide.md:
```diff
@@ -34,7 +34,7 @@ $ sudo mkdir -p /var/lib/teleport
 ### Teleport Checksum
 
-Gravitational Teleport provides a checksum from the 
+Gravitational Teleport provides a checksum from the
```
It's a bit of pain / makes it harder to review when removing trailing spaces, but I think it's a good practice and something we should write as part of our docs checks.

awly replied:
Ooops, didn't realize how many diffs that produced.
Reverted.

benarent commented on docs/4.2/architecture/teleport_auth.md:
```diff
@@ -218,41 +220,6 @@ storage.
-the proxy, users with root privileges on nodes have no way of disabling the
-audit.
-
-See the [admin guide](../admin-guide.md#recorded-sessions) to learn how to turn on the
```
Is this link still there in teleport_proxy.md ? We should also make clear that the setting is defined on the auth server vs proxy.

awly replied:
Yep, the link is there (the section is bit-to-bit identical).
Added a note in `teleport_proxy.md` that proxy recording is configured on auth server

benarent reviewed (changes requested):
Left a few comments, one change would be to make sure that we still link to the admin guide for how to setup recording mode.

Commit:
Minor architecture docs tweaks

- consistently use "certificate" instead of "public key"
- make diagram in "local users" section match the text (user "sandra"
  doesn't have access to "grav-02")
- de-duplicate docs on session streaming between auth and proxy pages

awly reviewed (commented):
PTAL

awly commented:
teleport_auth.md still links to https://gravitational.com/teleport/docs/admin-guide/#audit-log from "Audit Log" section, which has docs on configuring recording mode.
Do you think that's sufficient or should I add another, more direct link?

benarent reviewed (approved):
Looks good 👍

awly commented:
@gravitational-jenkins retest please

awly commented:
@gravitational-jenkins retest this please

awly commented:
@gravitational-jenkins retest this please

awly commented:
@gravitational-jenkins retest this please

awly commented:
@gravitational-jenkins retest this please
